src/nvim/main.c

```diff
@@ -217,10 +222,31 @@ void early_init(void)
 
 #ifdef MAKE_LIB
 int nvim_main(int argc, char **argv)
+#elif defined WIN32
+// don't use codepage encoded arguments. see #7060
+int wmain(int argc, wchar_t **argv_w)
 #else
 int main(int argc, char **argv)
 #endif
 {
+#ifdef WIN32
+  char *argv[argc];
+
+  for(size_t i = 0; i < (size_t)argc; i++) {
+    // get required buffer size
+    size_t dest_size = (size_t)WideCharToMultiByte(CP_UTF8, 0, argv_w[i], -1, NULL, 0, NULL, NULL);
+    char *buf = (char *)malloc(dest_size + 1);
```

Comment:
This can be changed to

```c
char *buf = xmallocz(dest_size);
```

It will either allocate the required memory or exit.  `xmallocz` also ensures the last byte is `NUL`.